src/ObjCRuntime/Protocol.cs

```diff
@@ -24,6 +24,23 @@ public Protocol (string name)
 				throw new ArgumentException (String.Format ("'{0}' is an unknown protocol", name));
 		}
 
+		public Protocol (Type type)
+		{
+			var exception = new ArgumentException (string.Format ("'{0}' is an unknown protocol", type.FullName));
```

Comment:
Also having different exception messages for each condition gives you valuable information (which check failed?).